pkg/operator/starter_controller.go

```diff
@@ -88,6 +89,11 @@ func (c *Controller) sync(ctx context.Context, syncCtx factory.SyncContext) erro
 		c.controllersRunning = true
 	}
 
+	// If the precondition was valid, let the controllers start and return any error after.
```

Comment:
Note for reviewers:

This is needed to get as close as possible to how it was done with the legacy csi.
https://github.com/openshift/csi-driver-manila-operator/blob/release-4.18/pkg/controllers/manila/manila.go#L90-L137